sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -343,8 +343,74 @@ public static ColumnarBatch from(Table table, DataType[] colTypes) {
       if (!(dt.equals(DType.LIST))) {
         return false;
       }
-      try (ColumnViewAccess<T> tmp = cv.getChildColumnViewAccess(0)) {
-        DType tmpType = tmp.getDataType();
+      try (ColumnView tmp = cv.getChildColumnView(0)) {
+        DType tmpType = tmp.getType();
+        return tmpType.equals(DType.INT8) || tmpType.equals(DType.UINT8);
+      }
+    } else {
+      // Unexpected type
+      return false;
+    }
+  }
+
+  private static boolean typeConversionAllowed(ColumnView cv, DataType colType) {
```

Comment:
If we are ok with exposing the columnView of a ColumnVector then I can remove some of the redundant code I had to add to make it work, since now ColumnVector and ColumnView dont have any shared hierarchy.